### PR TITLE
Fix additional character appended to Calendar cells in various locales

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/4135

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/calendar/src/useCalendarCell.ts [hunk 1]
```diff
@@ -302,7 +302,7 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
     calendar: date.calendar.identifier
   });
 
-  let formattedDate = useMemo(() => cellDateFormatter.format(nativeDate), [cellDateFormatter, nativeDate]);
+  let formattedDate = useMemo(() => cellDateFormatter.formatToParts(nativeDate)[0].value, [cellDateFormatter, nativeDate]);
```
Comment (devongovett) [inline packages/@react-aria/calendar/src/useCalendarCell.ts:305]: I don't think we should assume that it is always the first part.

Comment (ktabors) [reply packages/@react-aria/calendar/src/useCalendarCell.ts:305]: Are you suggesting we check both values and pick the numeric one?

Comment (snowystinger) [reply packages/@react-aria/calendar/src/useCalendarCell.ts]: no, he's saying look through the array and find the day value
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts

File: packages/@react-aria/calendar/src/useCalendarCell.ts [hunk 2]
```diff
@@ -302,7 +302,7 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
     calendar: date.calendar.identifier
   });
 
-  let formattedDate = useMemo(() => cellDateFormatter.format(nativeDate), [cellDateFormatter, nativeDate]);
+  let formattedDate = useMemo(() => cellDateFormatter.formatToParts(nativeDate).filter(part => part.type === 'day')[0].value, [cellDateFormatter, nativeDate]);
```
Comment (devongovett) [inline packages/@react-aria/calendar/src/useCalendarCell.ts:305]:
```suggestion
  let formattedDate = useMemo(() => cellDateFormatter.formatToParts(nativeDate).find(part => part.type === 'day').value, [cellDateFormatter, nativeDate]);
```

**Review by ktabors — APPROVED** [review]

**Comment by devongovett** [comment]
I'm still somewhat concerned that this won't be correct in 100% of locales. How do we verify this is actually what we want?

**Comment by snowystinger** [comment]
> I'm still somewhat concerned that this won't be correct in 100% of locales. How do we verify this is actually what we want?

I'm following up on an internal thread with the intl team to make sure.

That said, I think things are now more correct than they were. Given the issue was raised by native speakers and the intl team. So I think we can move forward with this and either wait for a bug indicating that something still needs these characters or the intl team comes up with an example that would require them.

**Comment by devongovett** [comment]
Right, sure, it's probably correct for the locales we have tested, but modifying the way the formatter works might cause problems for other locales. I just want to be sure fixing the issue doesn't regress somewhere else. Date formatting is very complex and locale dependent, so relying on the formatter to do the work seemed to be the best solution to ensure it worked everywhere.

**Comment by devongovett** [comment]
For example, here is where the formatting for "day" is defined in Unicode CLDR: https://github.com/unicode-org/cldr/blob/1de5706c84688b6745eb5d43175a51ea87b5891c/common/main/ja.xml#L1488

**Comment by snowystinger** [comment]
Intl did a more in depth look. I can give you the link to the review internally. Here's the summary of the findings though

The extra characters are returned by useDateFormatter for 19 locales.
There were no use cases with the “compact” calendar view, where the extra characters are used.
But there is one use case (MacOS calendar - month view), where the calendar cells are larger, when the extra characters are used. But this usage seems to be optional, as it’s not seen with another similar use case (eg Japanese Outlook for Mac calendar).
It looks like by default the extra characters should not be displayed. Perhaps there could be an additional option in useCalendarCell for displaying the extra characters for large calendar cells

**Comment by gesposito** [comment]
👋 Thanks for fixing this, is there a release date available?

**Comment by snowystinger** [comment]
We're working on the release right now. Hopefully in the next week or two

**Pull request merged**